### ISSUE TITLE
Change use of Dict to Map per Elixir 1.2 soft deprecation

### ISSUE THREAD

**Issue by parkerl** [exercism/xelixir#120]
Per the 1.2 changelog https://github.com/elixir-lang/elixir/blob/v1.2.0/CHANGELOG.md

```
atbash-cipher/example.exs
28:    Dict.get(@key, character, character)

etl/etl.exs
10:  @spec transform(Dict.t) :: map()

etl/example.exs
12:      |> Dict.to_list

forth/example.exs
98:          do_eval(%{s | defs: Dict.put(s.defs, w, ws), input: r})

minesweeper/example.exs
34:        Dict.update(acc, c1, 1, &(&1+1))

nucleotide-count/dna.exs
29:  @spec histogram([char]) :: Dict.t

palindrome-products/example.exs
22:    Dict.update(dict, product,

palindrome-products/palindrome_products_test.exs
16:    assert (palindromes |> Dict.keys |> Enum.sort |> List.last) == 9
23:    assert (palindromes |> Dict.keys |> Enum.sort |> List.last) == 9009
30:    assert (palindromes |> Dict.keys |> Enum.sort |> hd) == 121
37:    assert (palindromes |> Dict.keys |> Enum.sort |> List.last) == 906609
44:    assert (palindromes |> Dict.keys |> Enum.sort |> hd) == 10201

parallel-letter-frequency/example.exs
33:    |> Enum.reduce(%{}, fn c, acc -> Dict.update(acc, c, 1, &(&1+1)) end)
38:      Dict.merge(acc, d, fn _, a, b -> a+b end)

parallel-letter-frequency/frequency.exs
9:  @spec frequency([String.t], pos_integer) :: Dict.t

test/test_helper.exs
6:|> Dict.get("problems")

word-count/example.exs
21:    Dict.update(counts, word, 1, &(&1 + 1))
```

**Comment by parkerl** [comment]
@Teapane you are killing it! Thanks!

**Comment by parkerl** [comment]
Closed by #123

**Comment by Teapane** [comment]
:tada: :zap: :smile:
